target/linux/layerscape/files/arch/arm64/boot/dts/freescale/traverse-ls1043s.dts

```diff
@@ -0,0 +1,339 @@
+
+	emc1704@4c {
+		compatible = "microchip,emc1704";
+		reg = <0x4c>;
+	};
+
+	pac1934@16 {
+		compatible = "microchip,pac1934";
+		reg = <0x16>;
+		/* Monitoring 3.3V, 5V, Vin/12V (voltage only), Vbat/RTC (voltage only) */
+		shunt-resistors = <4000 12000 0 0>;
+	};
+
+	pmic@8 {
+		compatible = "freescale,mc34vr500";
+		reg = <0x08>;
+	};
+};
+
+&i2c3 {
+	status = "okay";
+	i2c-switch@70 {
+		compatible = "nxp,pca9540";
+		#address-cells = <1>;
+		#size-ceels = <0>;
+		reg = <0x70>;
+
+		gigsfp_i2c: i2c@0 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <0>;
+		};
+		tensfp_i2c: i2c@1 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <1>;
+		};
+	};
+
+};
+
+/* sfp_eth5: sfp-eth5 {
+	compatible = "sff,sfp";
+	i2c-bus = <&gigsfp_i2c>;
+}; */
+
+&ifc {
+	status = "okay";
+	#address-cells = <2>;
+	#size-cells = <1>;
+	/* NOR, NAND Flashes and FPGA on board */
+	ranges = <0x0 0x0 0x0 0x7e800000 0x00010000>;
+
+		nand@1,0 {
+			compatible = "fsl,ifc-nand";
+			#address-cells = <1>;
+			#size-cells = <1>;
+			reg = <0x0 0x0 0x10000>;
+		};
+};
+&duart0 {
+	status = "okay";
+};
+
+&duart1 {
+	status = "okay";
+};
+
+#include "fsl-ls1043-post.dtsi"
+
+&fman0 {
+	EMAC0: ethernet@e0000 {
+		phy-handle = <&qsgmii_phy1>;
+		phy-connection-type = "qsgmii";
+		local-mac-address = [0A 00 00 00 00 01];
+	};
+
+	EMAC1: ethernet@e2000 {
+		phy-handle = <&qsgmii_phy2>;
+		phy-connection-type = "qsgmii";
+		local-mac-address = [0A 00 00 00 00 02];
+	};
+
+	EMAC2: ethernet@e8000 {
+		phy-handle = <&qsgmii_phy3>;
+		phy-connection-type = "qsgmii";
+		local-mac-address = [0A 00 00 00 00 03];
+	};
+
+	EMAC3: ethernet@ea000 {
+		phy-handle = <&qsgmii_phy4>;
+		phy-connection-type = "qsgmii";
+		local-mac-address = [0A 00 00 00 00 04];
+	};
+
+	/* SFP via AR8031 
+	 * We treat this as a fixed-link as the
+	 * AR8031 is hard-configured into
+	 * 1000BASE-X mode
+	 */
+	EMAC4: ethernet@e4000 {
+		/* phy-handle = <&rgmii_phy1>; */
+		phy-connection-type = "rgmii";
+		local-mac-address = [0A 00 00 00 00 05];
+		fixed-link {
+			speed = <1000>;
+			full-duplex;
+		};
+	};
+
+	/* Connection to Expansion (M.2) slot
+	 * Future WAN (i.e xDSL) plugin
+	 */
+	EMAC5: ethernet@e6000 {
+		phy-connection-type = "rgmii-id";
+		local-mac-address = [00 00 00 00 00 06];
+		fixed-link {
+			speed = <1000>;
+			full-duplex;
+		};
+	};
+
+	/* 10G SFP+ interface 
+	 * Configure as SGMII so it can work at
+	 * 1G (SerDes Protocol 0x3455).
+	 * This is forward compatible with 10G
+	 */
+	TENSFP: ethernet@f0000 {
+		status = "okay";
+		phy-connection-type = "sgmii";
+		fixed-link {
+			speed = <1000>;
+			full-duplex;
+		}; 
+	};
+
+	mdio@fc000 {
+		rgmii_phy1: ethernet-phy@2 {
+			reg = <0x2>;
+		};
+		qsgmii_phy1: ethernet-phy@4 {
+			reg = <0x4>;
+		};
+		qsgmii_phy2: ethernet-phy@5 {
+			reg = <0x5>;
+		};
+		qsgmii_phy3: ethernet-phy@6 {
+			reg = <0x6>;
+		};
+		qsgmii_phy4: ethernet-phy@7 {
+			reg = <0x7>;
+		};
+	};
+
+};
+
+/* The LS1043S does not expose the
```

Comment:
Because?

Reply:
It isn't possible to build a board that has both SATA and 10G ports (see https://traverse.com.au/docs/ls1043s/faq/). The device tree setting is to prevent the AHCI driver from enumerating.